--- lutok-lutok-0.5/m4/lua.m4.orig	2024-11-26 02:09:07.000000000 +0100
+++ lutok-lutok-0.5/m4/lua.m4	2024-11-27 19:34:41.919832785 +0100
@@ -34,6 +34,10 @@ dnl
 AC_DEFUN([KYUA_LUA], [
     lua_found=no
 
+    PKG_CHECK_MODULES([LUA], [lua >= 5.1],
+                      [lua_found=yes], [true])
+
+    if test "${lua_found}" = no; then
     for lua_release in ${LUA_VERSION:-5.4 5.3 5.2 5.1}; do
         if test "${lua_found}" = no; then
             PKG_CHECK_MODULES([LUA], [lua${lua_release} >= ${lua_release}],
@@ -43,13 +47,10 @@ AC_DEFUN([KYUA_LUA], [
             PKG_CHECK_MODULES([LUA], [lua-${lua_release} >= ${lua_release}],
                               [lua_found=yes], [true])
         fi
-        if test "${lua_found}" = no; then
-            PKG_CHECK_MODULES([LUA], [lua >= ${lua_release}],
-                              [lua_found=yes], [true])
-        fi
 
         test "${lua_found}" = no || break
     done
+    fi
 
     if test "${lua_found}" = no; then
         AC_PATH_PROGS([LUA_CONFIG], [lua-config], [unset])
